### ISSUE TITLE
Proposed Version 2.0.0 Roadmap

### ISSUE THREAD

**Issue by paragonie-scott** [defuse/php-encryption#71]
After #63 has been reviewed and is merged, we should do #34. My proposal for #34 is as follows:

1. The first N bytes will be reserved for a version tag header. (N = 4, for example.)
2. For `Crypto.php`, we should add a method called `legacyDecrypt()` to retrieve the plaintext of messages encrypted with the 1.x strategy. File encryption/decryption should always be tagged as of the 2.0.0 release, so a backwards-compatible strategy is unnecessary. We should not allow legacy encryption.

------------

That would solve the thorniest issue, from my perspective.

Afterwards, I would say that the following issues should be part of the 2.0.0 milestone:

* #5 - We can use `hash_pbkdf2()` with sane constants and provide a compatible polyfill for PHP 5.4 (see [php-future](https://github.com/sarciszewski/php-future/blob/bd6c91fb924b2b35a3e4f4074a642868bd051baf/src/Security.php#L52-L87) for example)
* #68 - When this library is stable, we can use `random_bytes()` in PHP 7 and a polyfill that follows best practices on PHP 5.x, with no added conditional statements.
* ~~#70 - Will make it better to write `try/catch` blocks.~~

**Comment by paragonie-scott** [comment]
* My Version 2.0.0 pull request is in #78 and should close the following tickets (not automatically however):
    * #34 
    * #39 
    * #76 

If my propsal #80 is accepted, we can encapsulate #5 in that functionality and close that one too. At that point, version 2.0.0 will require an instance of `\Defuse\Crypto\MasterKey`.

I might make this key piece a separate PR once 78 has been reviewed.

**Comment by defuse** [comment]
Closing this because it doesn't have a well-defined goal, and I just made a milestone for v2.0 which we can use to track things.
